progress/README.md

```diff
@@ -0,0 +1,256 @@
+# Progress
+
+## **Requests**
+
+|                  Route                   |                         Description                          |
+| :--------------------------------------: | :----------------------------------------------------------: |
+|        [GET /progress](#get-post)        | Retrieves progress entries based on the provided parameters. |
+|     [POST /progress](#post-progress)     |                Creates a new progress entry.                 |
+| [DELETE /progress/:id](#delete-progress) |       Deletes a progress entry with the specified ID.        |
+
+## **GET /progress**
+
+Retrieves progress entries based on the provided parameters.
+
+- **Params**
+  None
+- **Query**  
+  type (query parameter): Specifies the type of progress to filter (e.g., "task", "user").
+  id (query parameter): Specifies the ID of the progress entry to retrieve (optional).
```

Comment:
cant it be GET `/progress/{id}?type="USER"`

Reply:
I think if we are adding as path param then let us add both in the path param. as both of them is anyways needed. would be better with consistency. 
`/progress/:type/:id"

Reply:
Type is still a filter, it wont be meaningful to keep it as a URL path.